src/osd/osd_types.cc

```diff
@@ -4125,9 +4120,13 @@ ostream& operator<<(ostream& out, const pg_log_entry_t& e)
 
 // -- pg_log_dup_t --
 
-string pg_log_dup_t::get_key_name() const
+std::string pg_log_dup_t::get_key_name() const
 {
-  return "dup_" + version.get_key_name();
+  alignas(4) static const char prefix[] = { 'd', 'u', 'p', '_' };
+  alignas(4) char key[36];
+  memcpy(key, prefix, 4);
+  version.get_key_name(key + 4);
+  return std::string(key);
```

Comment:
According to http://www.cplusplus.com/reference/string/string/data/, it is defined. Or maybe you meant C++14 and further?